src/util/git-ref-resolver.js

```diff
@@ -38,39 +39,36 @@ class GitRefResolver {
 
   async resolve(): Promise<DefaultBranch | ResolvedSha | false> {
     const testFunctions = [
```

Comment:
My first motivation for the class was the multiple parameters (`version`, `config`, `refs`, ... possibly growing). I can refactor and pass all of these as an object argument, that should indeed be more efficient.